AVOutput/AVOutputTV.cpp

```diff
@@ -0,0 +1,7556 @@
+                                      "PictureMode", platformsupport, index);
+        if (ret == tvERROR_NONE)
+        {
+            if (localrange.size() == index.size())
+            {
+                for (unsigned int i = 0; i< localrange.size(); i++)
+                {
+                    supportedPictureModemap[localrange[i]] = stoi(index[i]);
+                    LOGINFO("fetch the picture[%s] index[%d] \n", localrange[i].c_str(), stoi(index[i]));
+                }
+            }
+        }
+        else
+        {
+            LOGERR("%s: Failed to fetch the picture index \n", __FUNCTION__);
+            return -1;
+        }
+
+        if (!localpq.empty()) localpq.clear();
+        if (!localformat.empty()) localformat.clear();
+        if (!localsource.empty()) localsource.clear();
+        if (!localrange.empty()) localrange.clear();
+        if(!index.empty()) index.clear();
+
+        ret = getParamsCaps(localrange, localpq, localformat, localsource,
+                                      "VideoFormat", platformsupport, index);
+        if (ret == tvERROR_NONE)
+        {
+            if (localrange.size() == index.size())
+            {
+                for (unsigned int i = 0; i< localrange.size(); i++)
+                {
+                    LOGINFO("fetch the format[%s] index[%d] \n", localrange[i].c_str(), stoi(index[i]));
+                    supportedFormatmap[localrange[i]] = stoi(index[i]);
+                }
+            }
+        }
+        else
+        {
+            LOGERR("%s: Failed to fetch the format index \n", __FUNCTION__);
+            return -1;
+        }
+
+        return 0;
+    }
+
+    int AVOutputTV::getPictureModeIndex(std::string pqparam)
+    {
+	int index = -1;
+        std::map<std::string, int> :: iterator it;
+
+	for(it = supportedPictureModemap.begin(); it != supportedPictureModemap.end(); it++)
+	{
+	    if (it->first == pqparam)
+	    {
+	        index = it->second;
+		break;
+	    }
+	}
+	return index;
+    }
+
+    int AVOutputTV::getSourceIndex(std::string pqparam)
+    {
+	int index = -1;
+        std::map<std::string, int> :: iterator it;
+
+	for(it = supportedSourcemap.begin(); it != supportedSourcemap.end(); it++)
+        {
+            if (it->first == pqparam)
+            {
+                index = it->second;
+                break;
+            }
+        }
+        return index;
+    }
+
+    int AVOutputTV::getFormatIndex(std::string pqparam)
+    {
+        int index = -1;
+        std::map<std::string, int> :: iterator it;
+
+	for(it =  supportedFormatmap.begin(); it !=  supportedFormatmap.end(); it++)
+        {
+            if (it->first == pqparam)
+            {
+                index = it->second;
+                break;
+            }
+        }
+        return index;
+    }
+
+    void AVOutputTV::NotifyVideoFormatChange(tvVideoHDRFormat_t format)
+    {
+        JsonObject response;
+        response["currentVideoFormat"] = getVideoFormatTypeToString(format);
```

Comment:
is this come from TVsetting ?

Reply:
Yes. It is a tvsettings API.
void TVMgr::NotifyVideoFormatChange(tvVideoHDRFormat_t format)